### PR TITLE
Fixes a number of things, driven by the github openapi spec

### PR DESCRIPTION
**Pull request by Baccata**
See https://raw.githubusercontent.com/github/rest-api-description/main/descriptions/ghes-3.7/ghes-3.7.json

* nulls in enumerations
* head digits after sanitation
* empty enum member names after sanitation

#### Todo

- [x] add unit tests

### CODE REVIEW

File: modules/openapi/src/internals/IModelToSmithy.scala [hunk 1]
```diff
@@ -254,11 +262,17 @@ final class IModelToSmithy(useEnumTraitSyntax: Boolean)
   }
 
   private def sanitizeMemberName(id: String): String = {
-    removeInvalidCharactersForName(
-      sanitizeForDigitStart(id).replaceAll("-", "_")
+    sanitizeForDigitStart(
```
Comment (daddykotex) [inline modules/openapi/src/internals/IModelToSmithy.scala:265]: lol good catch

File: modules/openapi/src/internals/IModelToSmithy.scala [hunk 2]
```diff
@@ -222,6 +225,11 @@ final class IModelToSmithy(useEnumTraitSyntax: Boolean)
           sanitizeMemberName(memberName.value.toString)
         )
     }
+
+    def jsonName: Option[String] = id match {
```
Comment (Baccata) [inline modules/openapi/src/internals/IModelToSmithy.scala:229]: this is unused, will remove

**Review by kubukoz — APPROVED** [review]

**Comment by daddykotex** [comment]
Can i add some tests and wrap this PR? Asking in case you've already started and have some changes that you have not pushed out yet

File: modules/openapi/tests/src/EnumSpec.scala [hunk 3]
```diff
@@ -108,6 +108,40 @@ final class EnumSpec extends munit.FunSuite {
     )
   }
 
+  test("enum - v1".only) {
```
Comment (lewisjkl) [inline modules/openapi/tests/src/EnumSpec.scala:111]:
```suggestion
  test("enum - v1") {
```

File: modules/openapi/src/internals/IModelToSmithy.scala (hunk 2 above)
Comment (lewisjkl) [reply modules/openapi/src/internals/IModelToSmithy.scala:229]: Still need to remove? @daddykotex

Comment (daddykotex) [reply modules/openapi/src/internals/IModelToSmithy.scala]: yes I did not see the note

**Review by lewisjkl — APPROVED** [review]

**Pull request merged**